### PR TITLE
[REVIEW] Remove deprecated RMM allocations from synchronization.cpp

### PR DESCRIPTION
**Pull request by harrism**
synchronization.cpp, used in benchmarks, still had calls to `RMM_ALLOC` and `RMM_FREE`. cuDF's version of this file had been updated to `use device_buffer`, but not the copy here in RMM. So this PR updates it. 

It also deletes the example "test" benchmark, which is unneeded.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: benchmarks/synchronization/synchronization.cpp [hunk 1]
```diff
@@ -44,10 +39,8 @@ cuda_event_timer::cuda_event_timer(benchmark::State& state,
 
     if (l2_cache_bytes > 0) {
       const int memset_value = 0;
-      int* l2_cache_buffer   = nullptr;
-      RMM_TRY(RMM_ALLOC(&l2_cache_buffer, l2_cache_bytes, stream));
-      RMM_CUDA_TRY(cudaMemsetAsync(l2_cache_buffer, memset_value, l2_cache_bytes, stream));
-      RMM_TRY(RMM_FREE(l2_cache_buffer, stream));
+      rmm::device_buffer l2_cache_buffer(l2_cache_bytes, stream);
```
Comment (trevorsm7) [inline benchmarks/synchronization/synchronization.cpp:42]: Can `device_buffer` allocation throw?

Comment (harrism) [reply benchmarks/synchronization/synchronization.cpp:42]: Probably. Why?

Comment (trevorsm7) [reply benchmarks/synchronization/synchronization.cpp:42]: All good, just checking

**Pull request merged**